src/facade/dragonfly_connection.cc

```diff
@@ -399,8 +400,7 @@ uint32_t Connection::GetClientId() const {
 }
 
 bool Connection::IsAdmin() const {
-  uint16_t admin_port = absl::GetFlag(FLAGS_admin_port);
-  return socket_->LocalEndpoint().port() == admin_port;
+  return static_cast<Listener*>(owner())->IsAdminInterface();
```

Comment:
Can we add `IsAdminInterface()` API to `ListenerInterface`?
I realize it's perhaps not clean, but having a downcast is messy and can lead to issues down the road

Reply:
`IsAdminInterface` is dragonfly specific. On the other side, facade::Connection is always issued by facade::Listener.

Reply:
I'm open to any suggestions of removing hardcoded downcasts :)
We could even template-ize Listener to have facade Listener return its own connection

Reply:
My **personal** opinion is that we should avoid casts if there is a reasonable design choice that allows it. And if not - we can have them and not feel bad about it. Specifically, any virtual inheritance implies the possibility of downcasts and it's fine.

Reply:
I don't like the cast either, if we start doing it too much I think we should find out a better way, but for now this was the shortest way I found to do this :)